htdocs/compta/stats/cabyprodserv.php

```diff
@@ -469,3 +475,31 @@
 // End of page
 llxFooter();
 $db->close();
+
+/**
+ *  Retourne un tableau d'ID des sous-categories de l'ID catégorie donnée
+ *
+ *  @param	int		$rootCatID	(ID catégorie racine)
+ *  @param	mysqli	$db	(SQL DB object)
+ *  @return array	IDs sous-catégories de $rootCatID
+ */
+function getNestedChildrenCats($rootCatID, $db)
+{
+    $childrenCategoriesSQL =  'SELECT * FROM llx_categorie c WHERE c.fk_parent = '.$rootCatID;
+    $childrenCategories = $db->query($childrenCategoriesSQL)->fetch_all(MYSQLI_ASSOC);
```

Comment:
I'm afraid this may not be compatible with other databases than mysql.
Can you try to replace code of this method with the call of existing method
$categorie->get_full_arbo()